packages/node-fetch-server/src/lib/fetch-incoming-message/index.ts

```diff
@@ -0,0 +1,513 @@
+import type { Blob } from 'node:buffer';
+import type { Socket } from 'node:net';
+import type { ReadableStream } from 'node:stream/web';
+import { Readable, finished } from 'node:stream';
+
+import type {
+  Request,
+  ReferrerPolicy,
+  RequestCache,
+  RequestCredentials,
+  RequestDestination,
+  RequestMode,
+  RequestRedirect,
+} from 'undici-types';
+import SuperHeaders from '@mjackson/headers';
+
+import { consumeBody, parseJSONFromBytes, utf8DecodeBytes } from './body.js';
+import { matchKnownFields, onError, isDisturbed, serializeUrl } from './util.js';
+
+const kAbortController = Symbol('abortController');
+
+export class FetchIncomingMessage extends Readable implements Request {
+  //#region Private Fields
+
+  #state!: any;
+
+  #headers = new SuperHeaders();
+
+  #signal!: AbortSignal;
+
+  #stream!: ReadableStream<Uint8Array>;
+
+  [kAbortController]: AbortController | undefined;
+
+  //#endregion Private Fields
+
+  constructor(public readonly socket: Socket) {
+    super(socket ? { highWaterMark: socket.readableHighWaterMark } : undefined);
+
+    // initialize fetch request state
+    // see https://fetch.spec.whatwg.org/#dom-request
+    // see https://github.com/nodejs/undici/blob/main/lib/web/fetch/request.js
+
+    // const ac = new AbortController();
+    // this.on('close', () => {
+    //   ac.abort();
+    // });
+
+    // let signal = ac.signal;
+    // this.#signal = ac.signal;
+
+    // if (signal.aborted) {
+    //   ac.abort(signal.reason);
+    // } else {
+    //   // Keep a strong ref to ac while request object
+    //   // is alive. This is needed to prevent AbortController
+    //   // from being prematurely garbage collected.
+    //   // See, https://github.com/nodejs/undici/issues/1926.
+    //   this[kAbortController] = ac;
+    // }
```

Comment:
Is this supposed to be uncommented? Seems like it, otherwise `request.signal` will be `undefined`.